### ISSUE TITLE
How to use addStateTool on different stacks？

### ISSUE THREAD

**Issue by Hynn01** [cornerstonejs/cornerstoneTools#1223]
### Prerequisites

* [yes] Which version are you using? (Is it latest?)
* [yes] Are you reporting to the correct repository?
* [yes,#358 and #409 are partly related] Did you search existing issues? (Were any related?)

### Description

I want to make a list(like [[(x1,y1,z1),(w1,h1)],[(x2,y2,z2),(w2,h2)],[(x3,y3,z3),(w3,h3)]]) showed by rectangleRoiTool in different stacks(z layer) automatically, instead of human adding the rectangle. Is it possible to implement?
I find these issues similar to what I want to implement, but they just showed how to use the rectangleRoiTool on one stack. I still have no idea how to use it on different stacks?
#358
#409

**Comment by Hynn01** [comment]
Actually now I find the api "cornerstoneTools.globalImageIdSpecificToolStateManager.addImageIdToolState", I think it can implement what I want to do. Is it the right way to deal with my problem?

**Comment by Hynn01** [comment]
However, when I follow below steps, an error occurs as the picture show.
const measurementData2= {
      visible: true,
      active: false,
      color: undefined,
      invalidated: false,
      handles: {
          end: {
            x: 134.76562499999997,
            y: 76.70898288488387,
            highlight: true,
            active: false,
          },
          start: {
            x: 116.60156249999997,
            y: 65.86913913488387,
            highlight: true,
            active: false,
          },
          initialRotation: 0,
          textBox:{
            active: false,
            hasMoved: false,
            movesIndependently: false,
            drawnIndependently: true,
            allowedOutsideImage: true,
            hasBoundingBox: true,
            x: 134.76562499999997,
            y: 71.28906100988387,
          },
          boundingBox:{
            width: 120.9033203125,
            height: 25,
            left: 469.99999999999994,
            top: 230.83332824707028,
          },
          uuid: "44d212aa-9d04-4543-b08e-1407cc433b71",
          cachedStats:{
            area: 196.89559936523438,
            count: 209,
            mean: 242.92344497607655,
            variance: 68.72141205558728,
            stdDev: 8.289837878727623,
            min: 229,
            max: 255,
            meanStdDevSUV: undefined,
          },
          unit: "",
      },

cornerstoneTools.globalImageIdSpecificToolStateManager.addImageIdToolState(this.state.stack.imageIds[0], 'RectangleRoi', measurementData2);
<img width="1058" alt="屏幕快照 2020-04-18 下午8 36 34" src="https://user-images.githubusercontent.com/26082974/79685260-ff064780-8269-11ea-89c6-f9e219c54b47.png">

Is there anyway that I can put in the  information of rectangle(coordinate, length, width) and generate a rectangle class or something?

**Comment by dannyrb** [comment]
I'm not 100% sure I understand what you're trying to accomplish. Do you want to "restore/set state for a rectangleROI" on a stack or a specific image? What do you consider to be a stack? Do you need assistance identifying/finding stacks and imageIds?

**Comment by Hynn01** [comment]
Yes, I want to add a rectangleROI on a specific image. （Sorry for my poor English expression...I use the word "stack" by mistake. ）

**Comment by haoxl3** [comment]
@Hynn01 hi, I met the same problem as you, did you achieve it?

**Comment by Hynn01** [comment]
@haoxl3 I change the source code a little bit, add a createNewMeasurement2 in class RectangleRoiTool ,  new a RectangleRoiTool, and then createNewMeasurement2. I don't know whether it is the proper way to implement, but it works for me lol.

**Comment by desprit** [comment]
Any update on this? Is it possible to use `restoreToolState` to add to a specific slice of the stack?

**Comment by jackmcosta32** [comment]
You can use the state manger (cornerstoneTools.getElementToolStateManager) to get the stack state by getting state object inside of it, or use the getImageIdToolState function to retrive information from a specific image. Then, you can use the same state manger to add tools back by using the addImageIdToolState function or functions alike available on the stateManager object.

**Comment by desprit** [comment]
@jackmcosta32 no, stack state doesn't tell you anything about which slice exactly tool is bound to.
